chain/vm/fvm.go

```diff
@@ -565,6 +583,13 @@ func (vm *FVM) ApplyImplicitMessage(ctx context.Context, cmsg *types.Message) (*
 		Duration:       duration,
 	}
 
+	if len(ret.EventsBytes) > 0 && vm.returnEvents {
```

Comment:
nit: switch the order of clauses here to align with the same check in `ApplyMessage()`